migration/src/audio.rs

```diff
@@ -282,21 +296,39 @@ impl AudioRes {
     /// Converts this [AudioRes] into an [AudioSlice] representing audio for an entire document
     ///
     /// Note: Documents cannot have parent tracks at this time.
-    pub fn into_document_audio(self) -> AudioSlice {
-        AudioSlice {
+    pub fn into_document_audio(self) -> Result<AudioSlice, anyhow::Error> {
+        let resource_url = self.audio_url.clone();
+        let audio_duration = self.audio_duration.clone();
+        let annotations = self.into_audio_slices();
+
+        if let Some(slices) = &annotations {
+            let last_slice = slices
+                .last()
+                .ok_or_else(|| anyhow::format_err!("Expected there to be at least one slice"))?;
```

Comment:
Ive noticed that sometimes our logging appears in unpredictable orders due to use of async, so I think adding some info identifying the problematic document, annotations, and/or audio file as well as the particular word index and/or annotation time/index would be helpful in resolving errors thrown here